File: 백준/Silver/10773. 제로/제로.py
import sys
r = sys.stdin.readline
sys.setrecursionlimit(10**6)

def solution():
    k = int(r().rstrip())
    stk = []
    for _ in range(k):
        num = int(r().rstrip())
        if num == 0: stk.pop()
        else: stk.append(num)
            
    print(sum(stk))
    
solution()

File: 백준/Silver/10773. 제로/제로.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine());
        Stack<Integer> stk = new Stack<>();
        
        for(int i=0; i<n; i++) {
            int cmd = Integer.parseInt(br.readLine());
            if(cmd == 0) stk.pop();
            else stk.add(cmd);
        }
        
        int sum = 0;
        for(int v : stk) {
            sum += v;
        }
        System.out.println(sum);
    }
}